config/confighttp/compression_test.go

```diff
@@ -233,6 +233,40 @@ func TestHTTPContentDecompressionHandler(t *testing.T) {
 	}
 }
 
+func TestHTTPContentCompressionNilRequestBody(t *testing.T) {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+	})
+
+	addr := testutil.GetAvailableLocalAddress(t)
+	ln, err := net.Listen("tcp", addr)
+	require.NoError(t, err, "failed to create listener: %v", err)
+	srv := &http.Server{
+		Handler: handler,
+	}
+	go func() {
+		_ = srv.Serve(ln)
+	}()
+	// Wait for the servers to start
+	<-time.After(10 * time.Millisecond)
+
+	serverURL := fmt.Sprintf("http://%s", ln.Addr().String())
+
+	req, err := http.NewRequest("GET", serverURL, nil)
+	require.NoError(t, err, "failed to create request to test handler")
```

Comment:
If this fails, you'll end up with your server still running. I think there are some other tests running the require in a defer block, which might make sense here as well.